### PR TITLE
Fix some 404s

### PR DESCRIPTION
**Pull request by davisonio**
This fixes some 404s by removing them or changing the link if it's moved to a new location or avaliable elsewhere

### CODE REVIEW

File: free-programming-books-es.md [hunk 1]
```diff
@@ -172,7 +171,6 @@
 
 ### Git
 
-* [Git Immersion en Español](http://gitimmersion.mx)
```
Comment (onebree) [inline free-programming-books-es.md]: Spanish repository - https://github.com/esparta/gitimmersion-spanish

File: free-programming-books-es.md [hunk 2]
```diff
@@ -255,7 +253,7 @@
 
 ### PHP
 
-* [Manual de estudio introductorio al lenguaje PHP procedural](http://www.cursosdeprogramacionadistancia.com/static/pdf/material-sin-personalizar-php.pdf) (PDF)
+* [Manual de estudio introductorio al lenguaje PHP procedural](https://docs.google.com/viewer?a=v&pid=sites&srcid=aW5ub3ZhdmlydHVhbC5vcmd8dGVjY2VpbHB2aXxneDoyMWVlODE1ZWM5OTY5Yjg2) (PDF)
```
Comment (onebree) [inline free-programming-books-es.md]: Please avoid Google Docs or Dropbox. Please find an official source (github repo, author website). If none can be found, then please point to: 
https://web.archive.org/web/20140209203630/http://www.cursosdeprogramacionadistancia.com/static/pdf/material-sin-personalizar-php.pdf

^ Although I am unsure about the copyright of this source -- @vhf

File: free-programming-books-zh.md [hunk 3]
```diff
@@ -578,8 +578,6 @@
 * [Django book 2.0](http://djangobook.py3k.cn/2.0/)
 * [Python 3 文档(简体中文) 3.2.2 documentation](http://docspy3zh.readthedocs.org/en/latest/)
 * [Python 中文学习大本营](http://www.pythondoc.com)
-* [深入 Python](http://woodpecker.org.cn/diveintopython/)
-* [深入 Python 3](http://woodpecker.org.cn/diveintopython3/)
```
Comment (onebree) [inline free-programming-books-zh.md]: Mirror repo - https://github.com/jiechic/diveintopython3

File: free-programming-books.md [hunk 4]
```diff
@@ -1029,7 +1028,7 @@ Original Source: [List of freely available programming books](http://web.archive
 * [Developer Workbooks](http://developer.force.com/workbook)
 * [Force.com Fundamentals](https://developer.salesforce.com/page/Force_Platform_Fundamentals)
 * [Force.com Platform Fundamentals: An Introduction to Custom Application Development in the Cloud](http://www.lulu.com/shop/salesforcecom/forcecom-platform-fundamentals/ebook/product-17381451.html)
-* [Heroku Postgres](http://media.developerforce.com/workbooks/HerokuPostgres_Workbooks_Web_Final.pdf) (PDF)
+* [Heroku Postgres](http://www.scribd.com/doc/256342154/herokupostgres-workbooks-web-final-13-pdf) (PDF)
```
Comment (onebree) [inline free-programming-books.md]: @vhf what are your thoughts on links from Scribd? I have visited the site, but not in a while, so am mainly unfamiliar with the content and licensing.

File: free-programming-books.md [hunk 5]
```diff
@@ -1137,7 +1136,7 @@ Original Source: [List of freely available programming books](http://web.archive
 * [Introduction to Haskell](http://www.seas.upenn.edu/~cis194/spring13/) - Brent Yorgey
 * [Learn Haskell Fast and Hard](http://yannesposito.com/Scratch/en/blog/Haskell-the-Hard-Way/)
 * [Learn You a Haskell for Great Good](http://learnyouahaskell.com) - Miran Lipovaca
-* [Natural Language Processing for the Working Programmer](http://nlpwp.org/book/)
```
Comment (onebree) [inline free-programming-books.md]: 1. The main page says that the book is unfinished, and is unlikely to be revisited by its authors. Should we even keep such a book on the list?
2. I think the fact the website was taken down shows disinterest by the authors -- another reason to take the book off the list.

File: free-programming-books.md [hunk 6]
```diff
@@ -1817,7 +1816,7 @@ For resources on Angular, Backbone, D3, Dojo, Ember, Express, jQuery, Knockout,
 * [Advanced R Programming](http://adv-r.had.co.nz) - Hadley Wickham
 * [Introduction to Probability and Statistics Using R](http://cran.r-project.org/web/packages/IPSUR/vignettes/IPSUR.pdf) - G. Jay Kerns (PDF)
 * [Learning Statistics with R](http://health.adelaide.edu.au/psychology/ccs/teaching/lsr/) - Daniel Navarro
-* [learnR](https://github.com/renkun-ken/learnR) - (Work in Progress) - Kun Ren
```
Comment (onebree) [inline free-programming-books.md]: I see that the license is CC0, but I am concerned because the author abandoned the project -- it only exists as a fork.

Comment (davisonio) [inline free-programming-books.md]: It was the most up-to-date fork I found. @renkun-ken would you be okay with us keeping the fork here?

File: free-programming-books.md (hunk 4 above)
Comment (davisonio) [inline free-programming-books.md]: There's also an issuu link here: http://issuu.com/harshitpandey/docs/herokupostgres_workbooks_web_final

File: free-programming-books-es.md [hunk 7]
```diff
@@ -92,7 +92,7 @@
 * [Diseño Ágil con TDD](http://www.carlosble.com/libro-tdd/?lang=es)
 * [Ingeniería de Software: Una Guía para Crear Sistemas de Información](http://www.wolnm.org/apa/articulos/Ingenieria_Software.pdf) (PDF)
 * [Metodologías Ágiles para el Desarrollo de Software](http://www.desarrolloweb.com/manuales/metodologias-agil-desarrollo-software.html)
-* [Scrum & Extreme Programming (para programadores)](http://www.cursosdeprogramacionadistancia.com/static/pdf/material-sin-personalizar-agile.pdf) (PDF)
+* [Scrum & Extreme Programming (para programadores)](hhttps://web.archive.org/web/20140209204645/http://www.cursosdeprogramacionadistancia.com/static/pdf/material-sin-personalizar-agile.pdf) (PDF)
```
Comment (davisonio) [inline free-programming-books-es.md]: Added this resource back again because I found an archive.org link

File: free-programming-books-es.md (hunk 1 above)
Comment (davisonio) [inline free-programming-books-es.md]: Great, added!

File: free-programming-books-zh.md (hunk 3 above)
Comment (davisonio) [inline free-programming-books-zh.md]: Added that one but unfortunately can't seem to find one for diveintopython2

File: free-programming-books.md (hunk 5 above)
Comment (davisonio) [inline free-programming-books.md]: I see what you mean, have removed.

File: free-programming-books-es.md (hunk 2 above)
Comment (davisonio) [inline free-programming-books-es.md]: Added the archive.org link instead

File: free-programming-books-zh.md (hunk 3 above)
Comment (onebree) [inline free-programming-books-zh.md]: I know, I could not find it either.

**Comment by vhf** [comment]
Thanks a lot for your efforts @davisonio and @onebree!

File: free-programming-books.md (hunk 4 above)
Comment (vhf) [inline free-programming-books.md]: I'd vote against all scribd links. What do you think?

Comment (onebree) [inline free-programming-books.md]: @vhf +1 to that notion. But I am unsure about issuu, mainly because I never heard of it before

Comment (vhf) [inline free-programming-books.md]: I didn't know issuu either. I think we should adopt the same policy for all the "uploading" platforms. Issuu, scribd, gdrive, dropbox, etc.

Some of the files uploaded on these platforms and shared here are certainly legitimate, but their legitimacy is much harder to check without having an account on all these platforms. Also, I don't see why an author or a copyright holder would decide to exclusively share on issuu or scribd, knowing that the platform will most probably monetize the content in some way.

Same for drive and dropbox without the monetization issue. Anyone could upload anything there.

File: free-programming-books.md (hunk 6 above)
Comment (onebree) [inline free-programming-books.md]: @vhf thoughts with this one?

Comment (vhf) [inline free-programming-books.md]: I think https://github.com/yuanlisky/learnR is more up-to-date. The author committed there as recently as June 2015. @davisonio could you please use this one?

File: free-programming-books-tr.md [hunk 8]
```diff
@@ -139,7 +139,7 @@
 
 * [AB2014 Ruby Programlama Dili](https://github.com/leylaKapi/AB2014-Ruby-Programlama-Dili/blob/master/Ruby_AB2014.md)
 * [Ruby](https://www.ruby-lang.org/tr/)
-* [Ruby 101](https://www.gitbook.io/book/vigo/ruby-101)
+* [Ruby 101](https://vigo.gitbooks.io/ruby-101/content/)
```
Comment (vhf) [inline free-programming-books-tr.md]: Could you please remove `content/`?

File: free-programming-books-zh.md [hunk 9]
```diff
@@ -630,7 +629,7 @@
 
 ### Swift
 
-* [《The Swift Programming Language》中文版](https://www.gitbook.io/book/numbbbbb/-the-swift-programming-language-)
+* [《The Swift Programming Language》中文版](https://numbbbbb.gitbooks.io/-the-swift-programming-language-/content/)
```
Comment (vhf) [inline free-programming-books-zh.md]: Could you please remove `content/`?

File: free-programming-books.md [hunk 10]
```diff
@@ -1528,7 +1526,7 @@ For resources on Angular, Backbone, D3, Dojo, Ember, Express, jQuery, Knockout,
 * [Google's Objective-C Style Guide](http://google-styleguide.googlecode.com/svn/trunk/objcguide.xml)
 * [Object-Oriented Programming with Objective-C](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/OOP_ObjC/OOP_ObjC.pdf) (PDF)
 * [Objective-C Succinctly, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/objective-c) (PDF, Kindle) *(Just fill the fields with any values)*
-* [Programming With Objective-C](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/ProgrammingWithObjectiveC.pdf) (PDF)
+* [Programming With Objective-C](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/Introduction/Introduction.html) (PDF)
```
Comment (vhf) [inline free-programming-books.md]: Could you please remove ` (PDF)`?

**Comment by vhf** [comment]
I reviewed all changes and asked for some minor changes. Please ping me when it's done, I'll merge it! :)

File: free-programming-books-tr.md (hunk 8 above)
Comment (davisonio) [inline free-programming-books-tr.md]: Ok, but it means it'll redirect to https://www.gitbook.com/book/vigo/ruby-101/details

Comment (vhf) [inline free-programming-books-tr.md]: Yep. I don't expect gitbook to change their URL structure anytime soon. But if and when they do it, they'll most probably modify `/details` or `/content` and not the "base" URL leading to the book.

This redirect is perfectly fine for me. If you think this is not the right decision, I'd be glad to hear your point of view! :)

File: free-programming-books.md (hunk 6 above)
Comment (davisonio) [inline free-programming-books.md]: Yep, done

File: free-programming-books.md (hunk 10 above)
Comment (davisonio) [inline free-programming-books.md]: Oops, done

File: free-programming-books.md (hunk 4 above)
Comment (davisonio) [inline free-programming-books.md]: For now I've added the archive.org link to this resource - hope that's okay.

File: free-programming-books-zh.md (hunk 9 above)
Comment (davisonio) [inline free-programming-books-zh.md]: Done

**Comment by davisonio** [comment]
@vhf ping :)

File: free-programming-books-es.md (hunk 7 above)
Comment (vhf) [inline free-programming-books-es.md]: Typo: `hhttps`

Comment (davisonio) [inline free-programming-books-es.md]: Fixed

**Pull request merged**